src/socket/cache.ts

```diff
@@ -0,0 +1,6 @@
+import {BaseProduct} from "../models/product";
+import {UsersProducts} from "../models/usersProducts";
+
+export const productsAmount: BaseProduct[] = [];
+
+export const usersProducts: UsersProducts = {};
```

Comment:
move any access to usersProducts here (like class) ... create methods like update product, remove product, getCartById etc ... so your usersProducts will be private